### PR TITLE
VisualEditor: Remove wide and full alignment CSS and alignment classes on the root container

### LINKED ISSUE #52163: Page/Post editors: When template has "Inner blocks use content width", editing looks broken
Consider the following Page template layout, which features the Post Content block inside a 2-column setup, where the columns block applies proper padding to ensure that things are nicely spaced out:

site editor with columns and paddings

Note in this case how the Post Content block has the option, "Inner blocks use content width" unchecked. As a result of this option being unchecked, the regular page editor looks entirely broken, with an unbounded main column and no padding:

no main column or padding

This layout is entirely buggy and is not a good experience. 

Going back to the Page Template, if the "Inner blocks use content width" option is checked, the final site layout would look incorrect due to the intent of the 2-column setup:

site editor with columns, padding, and content width toggled

However when this option is enabled, the page editor looks usable again, with its trademark center column for content:

page editor with main column

Here's a GIF showing toggling and untoggling, and its effect on the page/post editors:

content width

In other words, it seems like the unbounded page and post editors were _intentionally_ made to unbound their main column when Post Content has this property toggled or untoggled. The net result is a very buggy post/page editor experience. Perhaps the intent was to be closer to WYSIWYG, but so long as the _container_ that surrounds the Post Content blog is omitted in page or post editors, it will never by WYSIWYG, it will only ever be broken. So until that can be the case, **post and page editors should not be affected by the Inner blocks/content width property**.

### LINKED ISSUE #60311: Post Editor does not reflect changes in core/post-content layout
As I investigated https://github.com/WordPress/gutenberg/issues/60308, I ran into this layout quirk, where the post editor would not respect the template's layout of `default`, to span inner blocks fullwidth. 

Using a `core/post-content` block with a `default` layout, blocks in the Site Editor were properly fullwidth, as was the front end of the page. **The Post Editor however did not reflect those changes.**

I expected each of these to be the same result, appropriately reflecting default and constrained. 

https://github.com/WordPress/gutenberg/assets/1813435/6994e871-b403-4f70-8b75-5c63c7812888

### PR DESCRIPTION
**Pull request by andrewserong**
## What?

Fixes #52163. Note, this does _not_ fix https://github.com/WordPress/gutenberg/issues/60311 which had the opposite feedback. If this PR lands, we'll effectively be saying that #60311 is a "won't fix" 🙂

This PR proposes removing the wide and full alignment CSS on the root container of the visual editor. I'm happy to close out this PR if it isn't a desired path forward.

## Why?

This CSS was in place so that the post and page editors could more closely resemble how a post might look in the context of a template. However, ever since the editors were unified, it's possible for users to switch on template preview mode (or in the case of page editing in the site editor, switch it off). In #52163 the feedback / discussion is that displaying post content full width in the post and page editors isn't a WYSIWYG experience and can make the editor feel broken.

The suggestion there was to remove the behaviour that detects wide / full width alignment and attempts to display content to the wide / full sizes in the post editor. Note: for posts and pages that use templates that do use content width, wide and full alignments should still be available _within_ the post content, so for example setting a Cover or Group block to wide or full in regular templates should work as before.

## How?

* In the visual editor, remove the `alignCSS` rules that attempted to make the editor (with template preview switched off) look more WYSIWYG
* Remove the output of align classes on the container

## Testing Instructions

1. Set up a test site running wordpress-develop trunk or WP 6.7 RC 1
2. Activate TT5 theme
3. In the site editor, go to templates and go to Add New Template
4. From the list, select `Single item: Post`
5. Select to create it for All Posts
6. You should then see a list to select from patterns to create your template. For this particular case, select this one that's optimised for photos:

image

7. Select the Content block in this template, and toggle off "Inner blocks use content width" so that the content fills the container:

image

8. Open up the post editor and add some content to the post. The content area should be constrained to a center column when template preview mode is off (the default state):

image

9. If you toggle the template to show the preview, the content will display to the full size of the container as in the template:

image

---

### Before

Note: on `trunk`, in this case, the content would have displayed edge-to-edge in the post editor:

image

### CODE REVIEW

File: packages/editor/src/components/visual-editor/index.js [hunk 1]
```diff
@@ -272,7 +272,7 @@ function VisualEditor( {
 			'is-layout-flow': ! themeSupportsLayout,
 		},
 		themeSupportsLayout && postContentLayoutClasses,
-		align && `align${ align }`
+		align && ! [ 'wide', 'full' ].includes( align ) && `align${ align }`
```
Comment (youknowriad) [inline packages/editor/src/components/visual-editor/index.js]: Any reason to keep left/right/center alignments?

Comment (andrewserong) [reply packages/editor/src/components/visual-editor/index.js]: Mostly out of caution as post content blocks that use left/right alignment look quite different to a center aligned column and the reported issue was to do with wide/full alignments. Happy to remove this altogether if we don't want the left/right alignment.

I believe @jasmussen and @richtabor have more thoughts about the behaviour here, so I'll hold off on any updates until they get a chance to look at this.

From my perspective I think there are good arguments both ways between attempting to display some alignments in the abstracted (template preview off) mode, and skipping them entirely. I don't mind which way we go with it, so I'll leave this PR open for now until there's a chance for more feedback on the approach.

TL;DR — let me know if anyone would like to remove the left/right/center alignments here and I'll remove them 🙂

Comment (tellthemachines) [reply packages/editor/src/components/visual-editor/index.js]: I think it would make sense to leave left/right alignments because they don't cause problems 😅 and so we can have _some_ degree of WYSIWYG.

**Review by tellthemachines — APPROVED** [review]
This works well for me in testing and it's not a huge change, only caters to a few edge cases, so I think we can try it and see how we go!

**Comment by andrewserong** [comment]
Thanks for the review @tellthemachines! 🙇

I'll hold off on merging just for now and see if there's any further feedback from @jasmussen, @richtabor or @WordPress/gutenberg-design. If not, we can try merging this in tomorrow or early next week.

File: packages/editor/src/components/visual-editor/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/editor/src/components/visual-editor/index.js]: Personally, I'm for consistency here. Left/right alignments for instance cause the whole thing to be "float" displayed, which IMO doesn't matter in this context at all and may actually create bugs. 

So instead of a half baked WYSIWYG, I think we should rather embrace that this is an abstracted view of the post editor and not meant to be full WYSIWYG in terms of layout.

Comment (tellthemachines) [reply packages/editor/src/components/visual-editor/index.js]: Oh hang on, actually I don't think there would be right/left align classes here; the Content block should only have wide/full enabled. 

When we set a constrained layout to justify left/right it doesn't use alignright or alignleft classes (or floats), it uses `margin-left: auto` or `margin-right: auto`. So it should be fine to remove the alignments here altogether! It won't make any difference to content justification.

Comment (andrewserong) [reply packages/editor/src/components/visual-editor/index.js]: Sounds good! I'll remove the align classes here 👍

Comment (andrewserong) [reply packages/editor/src/components/visual-editor/index.js]: Done in https://github.com/WordPress/gutenberg/pull/66352/commits/6d8f1cfba9d1eed0536a0a4b14d035e638051251

Comment (youknowriad) [reply packages/editor/src/components/visual-editor/index.js]: thanks

**Comment by jasmussen** [comment]
Thanks for exploring!

Rich and I were able to connect on this, to find out more fundamentally what was at play here. It's complex, though already mostly captured in the discussion above. It's just to say, if Rich has thoughts here, that's important to veer towards.

High level, there's an issue at play, where we have a abstracted post/page editor, which in many themes is _mostly_ WYSIWYG. It is, certainly, and should be, affected by color, typography, and every other property we can supply it with, to give you a decent idea of what the end result is. 

But the abstracted editor is also just the title and the _Content_ block. Any ancestor containers present in the Single templates, a Group with variable paddings, nested Columns blocks, sidebars left or right or both, are stripped from this view. This makes it inevitable that in some themes, this abstracted view differs quite a lot from the theme design itself. Simply put, not all themes have a centered column and whitespace left and right.

As much as is reasonable and possible, it's best to veer towards as much WYSIWYG in the abstracted editor as we can. And still balance that against a good writing experience. What Rich and I found in investigating this, is that trunk Gutenberg, except for one edge case, actually does this, treading that careful line very well. 

The edgecase we found was if you assign a wide or full-wide alignment to the Content block when it's inside a Group that defines content widths, and _then_ move the Content block outside of that. In this case, the wide or fullwide attribute still exists, but the button to toggle the attribute off is hidden, and yet the attribute is treated as if the element should be full-wide.

This edgecase, I guess, is technically a bug. It _seems_ like this PR fixes that, if that's the case, great. But otherwise, I was actually going to close #52163 and open a new one just for this edgecase. 

Let me know!

**Comment by andrewserong** [comment]
Thanks for laying that out @jasmussen!

> This edgecase, I guess, is technically a bug. It seems like this PR fixes that, if that's the case, great

Yes, this PR fixes that edge case. It also means that any Content block set in a template that has Full or Wide alignment attached to it will not have that class name attached to it in the abstracted editor. The more that I think about it, the more I believe this PR is probably the right way to go, as I think that effort to display `wide` or `full` was likely to be buggy, as you encountered.

So I'm still in favour of merging this PR. For templates/themes that set the Content block at the very root of the document and use flow/default layout, the editor will still display edge-to-edge for that use case of Page templates that are intended to be full wide to the very edge of the screen.

Also with the idea you've raised in https://github.com/WordPress/gutenberg/issues/66429 to better expose the toggle to preview the template, I think we're in a good place to proceed with this PR.

Let me know if anyone has any objections to landing this, and if not, I'll intend to press "Squash and merge" on Monday 🙂

**Comment by jasmussen** [comment]
I've no objection to merging. Though if Rich gets a chance (unclear he will before Monday), I'd appreciate, just so we don't regress some nuance in some of the behavior, where for example an image that was actually intentionally set to full-wide doesn't stop working inside this abstracted Content block, or the likes. I'll drop a link in case he gets a chance to look.

**Comment by andrewserong** [comment]
Thanks for the update, Joen! I'll leave this PR open until @richtabor has a chance to give it a look. Since this fixes something that's a bit of an edge case, I'm happy to leave this open for a while as it doesn't seem urgent.
